src/flow.jl

```diff
@@ -1102,6 +1145,287 @@ function update(sw::ShallowWaterLand{T}, swr::ShallowWaterRiver{T}, network, Δt
     end
 end
 
+""" 
+    FloodPlain
+
+Floodplain `volume` and mannings roughness `n` are a function of `depth` (flood depth
+intervals). Based on the floodplain `volume` a floodplain geometry as a function of
+`flood_depth` is derived with floodplain area `a` (cumulative), wetted perimeter radius `p`
+(cumulative), side `slope` (rectangular or trapezoidal) and rescaled wetted perimeter
+`p_unit` by `depth`.
+"""
+@get_units @with_kw struct FloodPlain{T,N}
+    mannings_n::Vector{SVector{N,T}} | "s m-1/3"        # Manning's roughness
+    depth::Vector{T} | "m"                              # Flood depth
+    volume::Vector{SVector{N,T}} | "m3"                 # Flood volume (cumulative)
+    width::Vector{SVector{N,T}} | "m"                   # Flood width
+    a::Vector{SVector{N,T}} | "m2"                      # Flow area (cumulative)
+    p::Vector{SVector{N,T}} | "m"                       # Wetted perimeter (cumulative)
+    p_unit::Vector{SVector{N,T}} | "-"                  # Wetted perimeter per unit flood depth
+    slope::Vector{SVector{N,T}} | "-"                   # Slope of floodplain
+end
+
+"helper function to get interpolation indices"
+function interpolation_indices(x, v::AbstractVector)
+    i1 = 1
+    for i in eachindex(v)
+        if v[i] <= x
+            i1 = i
+        end
+    end
+    if i1 == length(v)
+        i2 = i1
+    else
+        i2 = i1 + 1
+    end
+    return i1, i2
+end
+
+"helper function to get nearest neighbor index"
+function nearest_neighbor_index(x, v::AbstractVector)
+    idx = 0
+    Δxₘᵢₙ = Inf
+    for i in eachindex(v)
+        Δx = abs(v[i] - x)
+        idx = Δx < Δxₘᵢₙ ? i : idx
+        Δxₘᵢₙ = Δx < Δxₘᵢₙ ? Δx : Δxₘᵢₙ
+
+    end
+    return idx
+end
+
+"Compute river (and floodplain) mannings roughness value at edge `i` from adjacent nodes `i_src` and `i_dst` with wetted perimeter `p_src` and `p_dst` respectively"
+function interpolate_roughness_nodes(
+    sw::ShallowWaterRiver,
+    i::Int,
+    i_src::Int,
+    i_dst::Int,
+    p_src::Real,
+    p_dst::Real,
+)
+
+    if sw.hf[i] <= sw.bankfull_depth[i_src]
+        n_src = sw.mannings_n[i_src]
+    else
+        flood_depth = sw.hf[i] - sw.bankfull_depth[i_src]
+        j = max(nearest_neighbor_index(flood_depth, sw.floodplain.depth), 2)
```

Comment:
Why is this value maximized at 2? It is used to select the correct manning value, which should be defined at the same intervals as the `sw.floodplain.depth` values right?

Reply:
This is because index 1 is at flood depth 0.0 (convenient for interpolation), without a manning's n value for the floodplain.